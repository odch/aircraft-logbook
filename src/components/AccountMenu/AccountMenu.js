import React from 'react'
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'
import { FormattedMessage } from 'react-intl'
import MenuItem from '@material-ui/core/MenuItem'
import Menu from '@material-ui/core/Menu'
import Divider from '@material-ui/core/Divider'
import organizationShape from '../../shapes/organization'

class AccountMenu extends React.Component {
  render() {
    return (
      <Menu
        anchorEl={this.props.anchorEl}
        anchorOrigin={{
          vertical: 'top',
          horizontal: 'right'
        }}
        transformOrigin={{
          vertical: 'top',
          horizontal: 'right'
        }}
        open={this.props.open}
        onClose={this.props.onClose}
      >
        <MenuItem data-cy="menu-item-username" disabled>
          {this.props.auth.email}
        </MenuItem>
        <Divider key="username-divider" />
        {this.props.organization && [
          <MenuItem
            data-cy="menu-item-selected-organization"
            component={Link}
            to={`/organizations/${this.props.organization.id}`}
            key={`menu-item-selected-organization-${
              this.props.organization.id
            }`}
          >
            {this.props.organization.id}
          </MenuItem>,
          <MenuItem
            data-cy="menu-item-selected-organization-settings"
            component={Link}
            to={`/organizations/${this.props.organization.id}/settings`}
            key={`menu-item-selected-organization-settings-${
              this.props.organization.id
            }`}
          >
            <FormattedMessage id="menu.account.organizationsettings" />
          </MenuItem>,
          <Divider key="selected-organization-divider" />
        ]}
        <MenuItem
          data-cy="menu-item-organizations"
          component={Link}
          to="/organizations"
        >
          <FormattedMessage id="menu.account.organizations" />
        </MenuItem>
        <MenuItem data-cy="menu-item-logout" onClick={this.props.logout}>
          <FormattedMessage id="login.logout" />
        </MenuItem>
      </Menu>
    )
  }
}

AccountMenu.propTypes = {
  auth: PropTypes.shape({
    email: PropTypes.string.isRequired
  }).isRequired,
  organization: organizationShape,
  open: PropTypes.bool.isRequired,
  anchorEl: PropTypes.object,
  logout: PropTypes.func.isRequired,
  onClose: PropTypes.func.isRequired
}

export default AccountMenu
