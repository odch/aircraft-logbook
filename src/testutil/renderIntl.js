import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
