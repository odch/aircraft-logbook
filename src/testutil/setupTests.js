import * as enzyme from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import moment from 'moment'
import momentTimezone from 'moment-timezone'
import { init as initFeatureToggles } from '../util/featureToggles'

enzyme.configure({ adapter: new Adapter() })

moment.locale('de')
momentTimezone.locale('de')

initFeatureToggles()
