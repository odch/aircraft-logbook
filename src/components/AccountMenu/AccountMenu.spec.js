import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import AccountMenu from './AccountMenu'
import { renderIntlMaterial } from '../../testutil/renderIntl'

describe('components', () => {
  describe('AccountMenu', () => {
    it('does not render if not open', () => {
      const tree = renderIntlMaterial(
        <AccountMenu
          auth={{
            email: 'test@opendigital.ch'
          }}
          open={false}
          logout={() => {}}
          onClose={() => {}}
        />
      ).html()
      expect(tree).toMatchSnapshot()
    })

    it('renders without organization if no one is given', () => {
      const tree = renderIntlMaterial(
        <Router>
          <AccountMenu
            auth={{
              email: 'test@opendigital.ch'
            }}
            open
            logout={() => {}}
            onClose={() => {}}
          />
        </Router>
      ).html()
      expect(tree).toMatchSnapshot()
    })

    it('renders with organization if one is given', () => {
      const tree = renderIntlMaterial(
        <Router>
          <AccountMenu
            auth={{
              email: 'test@opendigital.ch'
            }}
            open
            organization={{
              id: 'my_org'
            }}
            logout={() => {}}
            onClose={() => {}}
          />
        </Router>
      ).html()
      expect(tree).toMatchSnapshot()
    })
  })
})
