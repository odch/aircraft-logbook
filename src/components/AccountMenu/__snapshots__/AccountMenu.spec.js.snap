// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components AccountMenu does not render if not open 1`] = `null`;

exports[`components AccountMenu renders with organization if one is given 1`] = `"<div class=\\"MuiModal-root-3\\"><div class=\\"MuiBackdrop-root-5 MuiBackdrop-invisible-6\\" aria-hidden=\\"true\\" style=\\"opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div class=\\"MuiPaper-root-7 MuiMenu-paper-1 MuiPaper-elevation8-17 MuiPaper-rounded-8 MuiPopover-paper-2\\" style=\\"opacity: 1; transform: scale(1, 1); top: 16px; left: 16px;\\" role=\\"document\\" tabindex=\\"-1\\"><ul class=\\"MuiList-root-34 MuiList-padding-35\\" role=\\"menu\\"><li tabindex=\\"-1\\" class=\\"MuiButtonBase-root-51 MuiButtonBase-disabled-52 MuiListItem-root-40 MuiListItem-default-43 MuiListItem-gutters-47 MuiListItem-disabled-45 MuiListItem-button-48 MuiMenuItem-root-38\\" role=\\"menuitem\\" data-cy=\\"menu-item-username\\">test@opendigital.ch</li><hr class=\\"MuiDivider-root-54\\" tabindex=\\"-1\\"><a tabindex=\\"-1\\" class=\\"MuiButtonBase-root-51 MuiListItem-root-40 MuiListItem-default-43 MuiListItem-gutters-47 MuiListItem-button-48 MuiMenuItem-root-38\\" role=\\"menuitem\\" data-cy=\\"menu-item-selected-organization\\" href=\\"/organizations/my_org\\">my_org<span class=\\"MuiTouchRipple-root-58\\"></span></a><a tabindex=\\"-1\\" class=\\"MuiButtonBase-root-51 MuiListItem-root-40 MuiListItem-default-43 MuiListItem-gutters-47 MuiListItem-button-48 MuiMenuItem-root-38\\" role=\\"menuitem\\" data-cy=\\"menu-item-selected-organization-settings\\" href=\\"/organizations/my_org/settings\\"><span>Organisationseinstellungen</span><span class=\\"MuiTouchRipple-root-58\\"></span></a><hr class=\\"MuiDivider-root-54\\" tabindex=\\"-1\\"><a tabindex=\\"-1\\" class=\\"MuiButtonBase-root-51 MuiListItem-root-40 MuiListItem-default-43 MuiListItem-gutters-47 MuiListItem-button-48 MuiMenuItem-root-38\\" role=\\"menuitem\\" data-cy=\\"menu-item-organizations\\" href=\\"/organizations\\"><span>Meine Organisationen</span><span class=\\"MuiTouchRipple-root-58\\"></span></a><li tabindex=\\"-1\\" class=\\"MuiButtonBase-root-51 MuiListItem-root-40 MuiListItem-default-43 MuiListItem-gutters-47 MuiListItem-button-48 MuiMenuItem-root-38\\" role=\\"menuitem\\" data-cy=\\"menu-item-logout\\"><span>Abmelden</span><span class=\\"MuiTouchRipple-root-58\\"></span></li></ul></div></div>"`;

exports[`components AccountMenu renders without organization if no one is given 1`] = `"<div class=\\"MuiModal-root-3\\"><div class=\\"MuiBackdrop-root-5 MuiBackdrop-invisible-6\\" aria-hidden=\\"true\\" style=\\"opacity: 1; -webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;\\"></div><div class=\\"MuiPaper-root-7 MuiMenu-paper-1 MuiPaper-elevation8-17 MuiPaper-rounded-8 MuiPopover-paper-2\\" style=\\"opacity: 1; transform: scale(1, 1); top: 16px; left: 16px;\\" role=\\"document\\" tabindex=\\"-1\\"><ul class=\\"MuiList-root-34 MuiList-padding-35\\" role=\\"menu\\"><li tabindex=\\"-1\\" class=\\"MuiButtonBase-root-51 MuiButtonBase-disabled-52 MuiListItem-root-40 MuiListItem-default-43 MuiListItem-gutters-47 MuiListItem-disabled-45 MuiListItem-button-48 MuiMenuItem-root-38\\" role=\\"menuitem\\" data-cy=\\"menu-item-username\\">test@opendigital.ch</li><hr class=\\"MuiDivider-root-54\\" tabindex=\\"-1\\"><a tabindex=\\"-1\\" class=\\"MuiButtonBase-root-51 MuiListItem-root-40 MuiListItem-default-43 MuiListItem-gutters-47 MuiListItem-button-48 MuiMenuItem-root-38\\" role=\\"menuitem\\" data-cy=\\"menu-item-organizations\\" href=\\"/organizations\\"><span>Meine Organisationen</span><span class=\\"MuiTouchRipple-root-58\\"></span></a><li tabindex=\\"-1\\" class=\\"MuiButtonBase-root-51 MuiListItem-root-40 MuiListItem-default-43 MuiListItem-gutters-47 MuiListItem-button-48 MuiMenuItem-root-38\\" role=\\"menuitem\\" data-cy=\\"menu-item-logout\\"><span>Abmelden</span><span class=\\"MuiTouchRipple-root-58\\"></span></li></ul></div></div>"`;
